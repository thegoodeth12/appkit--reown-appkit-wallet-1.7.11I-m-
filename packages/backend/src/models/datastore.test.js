import { describe, it, expect } from 'vitest';
import Datastore from '@/models/datastore.js';

describe('Datastore model', () => {
  it('tableName should return correct name', () => {
    expect(Datastore.tableName).toBe('datastore');
  });

  it('jsonSchema should have correct validations', () => {
    expect(Datastore.jsonSchema).toMatchSnapshot();
  });
});
