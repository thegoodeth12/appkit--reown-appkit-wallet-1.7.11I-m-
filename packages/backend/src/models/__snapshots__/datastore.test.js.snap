// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Datastore model > jsonSchema should have correct validations 1`] = `
{
  "properties": {
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "key": {
      "minLength": 1,
      "type": "string",
    },
    "scope": {
      "default": "flow",
      "enum": [
        "flow",
      ],
      "type": "string",
    },
    "scopeId": {
      "format": "uuid",
      "type": "string",
    },
    "value": {
      "type": "string",
    },
  },
  "required": [
    "key",
    "value",
    "scopeId",
  ],
  "type": "object",
}
`;
